### PR TITLE
🐛 Return an error if we can't delete the external remediation object

### PR DESCRIPTION
**Pull request by goushicui**
**What this PR does / why we need it**:
return delete remediation object error triger next reconcile

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: controllers/machinehealthcheck_controller.go [hunk 1]
```diff
@@ -347,7 +347,7 @@ func (r *MachineHealthCheckReconciler) patchHealthyTargets(ctx context.Context,
 			if obj.GetDeletionTimestamp() == nil {
 				// Issue a delete for remediation request.
 				if err := r.Client.Delete(ctx, obj); err != nil && !apierrors.IsNotFound(err) {
-					logger.Error(err, "failed to delete %v %q for Machine %q", obj.GroupVersionKind(), obj.GetName(), t.Machine.Name)
+					errList = append(errList, errors.Wrapf(err, "failed to delete %v %q for Machine %q", obj.GroupVersionKind(), obj.GetName(), t.Machine.Name))
```
Comment (killianmuldoon) [inline controllers/machinehealthcheck_controller.go:350]: @goushicui Can you explain why you'd like this to return here? Did you identify some unexpected behaviour with the current version?

Comment (enxebre) [reply controllers/machinehealthcheck_controller.go:350]: If we did this, wouldn't we want to only append if error != notFound?

**Comment by enxebre** [comment]
@sedefsavas can you recall why the error was ignored in the first place? https://github.com/kubernetes-sigs/cluster-api/pull/3713/files#diff-34cc13ef9d81a69fe97139ffce42c13fbf1ce1d330b00db41f9b29fcde9baaa4R310-R313

File: controllers/machinehealthcheck_controller.go (hunk 1 above)
Comment (goushicui) [reply controllers/machinehealthcheck_controller.go:350]: yes, just a choice that waiting nornal change machine or node update  to triger next reconceil or return error to tiger controller error handler?

Comment (sbueringer) [reply controllers/machinehealthcheck_controller.go:350]: > If we did this, wouldn't we want to only append if error != notFound?

I missed it too, but that check is already in l.349.

**Comment by goushicui** [comment]
@vincepri if e.g deleting the remediation request fails here, I think we should not patch machine MachineHealthCheckSuccededCondition as true

**Comment by stmcginnis** [comment]
/retest

Let's get some fresh test logs.

This makes sense to me. It would be interesting to know if there was a reason to only log the delete failures though. I don't see why, but it's possible there was some logic behind it.

**Comment by vincepri** [comment]
/retitle 🐛 Return an error if we can't delete the external remediation object

**Comment by fabriziopandini** [comment]
I'm +1 with the change, the error gets logged with the aggregated error and we are triggering requeue to retry the operation
/lgtm

**Pull request merged**